atom/browser/api/atom_api_session.cc

```diff
@@ -564,15 +565,17 @@ void Session::SetPermissionCheckHandler(v8::Local<v8::Value> val,
   permission_manager->SetPermissionCheckHandler(handler);
 }
 
-void Session::ClearHostResolverCache(mate::Arguments* args) {
-  base::Closure callback;
-  args->GetNext(&callback);
+v8::Local<v8::Promise> Session::ClearHostResolverCache() {
+  v8::Isolate* isolate = v8::Isolate::GetCurrent();
+  util::Promise promise(isolate);
```

Comment:
Keep `Session::ClearHostResolverCache(mate::Arguments* args)` signature and prefer `args->isolate()`